pil-analyzer/src/type_inference.rs

```diff
@@ -58,6 +59,8 @@ struct TypeChecker<'a> {
     /// Declared types for all symbols. Contains the unmodified type scheme for symbols
     /// with generic types and newly created type variables for symbols without declared type.
     declared_types: HashMap<String, TypeScheme>,
+    /// Source references for all declared types.
+    declared_type_source_refs: HashMap<String, SourceRef>,
```

Comment:
is this supposed to have the same keys as `declared_types`? If yes, these could be a tuple here

Reply:
Done.